### ISSUE TITLE
Spring ArmeriaAutoConfiguration could look at spring server settings too

### ISSUE THREAD

**Issue by anuraaga** [line/armeria#1834]
Many of the knobs for controlling an armeria server using spring integration are under the `armeria` configuration namespace, often to give more flexibility to configure armeria (e.g., ports). But to make migrating an existing Spring server to armeria simpler, does it make sense to fallback to also looking at `server` configuration.

Reference: https://github.com/apache/incubator-zipkin/issues/2627

**Comment by trustin** [comment]
I think it's doable if we set clear policy on what to do when a user specifies both `server` and `armeria` section. So far, we did not look into the `server` section because of ambiguity IIRC. Am I correct, @hyangtack ?

**Comment by hyangtack** [comment]
Yes, it would be possible, but I didn't do it because of the ambiguity between Spring Boot 2.x and 1.x, e.g. different package name, different class type for data unit, etc. However, I didn't look into it deeply at that time, so it'd be possible to find a simple way to do it. :-)

**Comment by adriancole** [comment]
agree that doing this implies maintenance when spring boot minor and major versions come out.

**Comment by trustin** [comment]
@adriancole wrote:

> ex ideal world is I can paste something like this and no-one is the wiser that armeria is even in use. While it is nice that it is, for some beginners it won't do anything except confuse to speak at this abstraction https://www.baeldung.com/spring-boot-https-self-signed-certificate

(Copied from a Slack thread)

**Comment by minwoox** [comment]
We can start this by making `AbstractArmeriaAutoConfiguration` (with a different name) to extend `AbstractConfigurableWebServerFactory`. 
Currently, it's not possible because `AbstractArmeriaAutoConfiguration` is used for boot1 and boot2.
We need to extract the common logic to a utility class and use them together in boot1 and boot2.
